Diagnostics/Get-NetView.PS1

```diff
@@ -1556,23 +1554,42 @@ function Environment {
     }
 } # Environment()
 
-function EnvDestroy {
+function EnvSetup {
     [CmdletBinding()]
     Param(
-        [parameter(Mandatory=$true)] [String] $OutDir
+        [parameter(Mandatory=$false)] [String] $OutputDirectory,
+        [parameter(Mandatory=$true)] [String] $WorkingDirectoryName
     )
-    If (Test-Path $OutDir) {
-        Remove-Item $OutDir -Recurse
+
+    # Process/Validate $OutputDirectory
+    function NormalizeOutDir {
+        if ($OutputDirectory) {
+            if (Test-Path $OutputDirectory) {
+                return (Resolve-Path $OutputDirectory).Path # full path
+            } else {
+                Throw "Get-NetView: The directory '$OutputDirectory' does not exist."
+            }
+        } # else $OutputDirectory is undefined
+        return [Environment]::GetFolderPath("Desktop") + "\" 
+    } # NormalizeOutDir ()
+
+    # Calculate working directory
+    $workDir = Join-Path $(NormalizeOutDir) $WorkingDirectoryName
+
+    # Delete working directory (and all of its contents) if it already exists
+    If (Test-Path $workDir) {
+        Remove-Item $workDir -Recurse
```

Comment:
No, please keep create() separate from destroy(), merging the two introduces unnecessary maintenance complexities.  As an example, the previous implementation was self descriptive and no need to comment.  The current implementation is overloaded in that it does two things, both create and destroy which goes against cohesion principles, requires comments to understand, and implicitly does destroy/create encompassed in setup.  You've also introduced a nested function which is not ideal.  It's extremely important that this script remain simple and easy to maintain.  Lastly, I don't follow why this change is required due to $workDir?  It was an input parameter in the previous function, why is it a problem which warrants this refactoring?  It was already grouped with other script wide vars such as $columns and $version.